maproulette/maproulette2
# merge to dev

* Support for Virtual Projects (#550) (#551)

* Create evolution to add virtual project_challenge join table
  and project.is_virtual column

* Create virtual project controller/dal to handle add/remove challenges

* Modify project and challenge queries to use new join table

* Add timestamp column to user_leaderboard (#558)

* Add timestamp column 'created' to user_leaderboard

* In api return created field in json

* Bug where only reviewers can contest their rejected tasks (#557)

* Allow users to contest their rejected tasks even if they
  aren't a reviewer

* Fix review status not updating in cache in setTaskStatus (#556)

* Add virtual projects to challenges listing project filter (#555)

* Prevent task loop at priority boundary (#554)

* Always include check to prevent randomly serving up tasks that the
user has recently worked on, rather than only performing the check when
requesting proximate tasks, as the same sort of task loop can occur when
a user is skipping the last task at a priority level

* Add additional support for virtual projects in data methods (#553)

* Also add Postman tests for virtual projects to ensure methods
  run without errors.

* Deprecated method getChallengeSummaries as it is not used
  and does not support virtual projects.

* Add support for all reviewed tasks (#564)

* CSV export bug -- reviewer column was wrong (#563)

* Allow project name in extendedFind search (#562)

* Support project name (`ps`) in extendedFind challenge searches to
further narrow down challenge results

* Change project-name search to use display_name instead of name

* Fix missing space if project-name search clause was added to an
existing WHERE clause

* Fix created column not present in non-prebuilt leaderboard queries (#561)

* Delete tasks asynchronously and in batches (#559)

* Add new DELETING_TASKS challenge status

* Change the `DELETE /challenge/:id/tasks` API endpoint to delete tasks
asynchronously and to change the challenge status to DELETING_TASKS
while the delete operation is ongoing

* Change `ChallengeDal.deleteTasks` to work in small batches

* Add guard conditions to ChallengeController to prevent bulk task
deletion or challenge rebuilds while a deletion or rebuild is already in
progress

* Change getChallengeGeometry to inclue task info. (#565)

* Add additional geoJSON properties field "maproulette"
  which includes additional MR related task info

* Fixed query when using status filter

* Added additional filter 'reviewStatus' and 'priority'
